app/application/controllers/app.rb

```diff
@@ -8,65 +8,57 @@
 module PaperDeep
   # Web App
   class App < Roda
-    use Rack::Cors, debug: true, logger: Logger.new($stdout) do
-      allowed_methods = %i[get post put delete options head]
-      allow do
-        origins 'localhost:3000'
-        resource '*', headers: :any, methods: allowed_methods, credentials: true
-      end
-    end
-    plugin :public, root: 'app/presentation/built', gzip: true
+    # use Rack::Cors, debug: true, logger: Logger.new($stdout) do
+    #   allowed_methods = %i[get post put delete options head]
+    #   allow do
+    #     origins 'localhost:3000'
+    #     resource '*', headers: :any, methods: allowed_methods, credentials: true
+    #   end
+    # end
+
     plugin :halt
-    plugin :flash
-    route do |routing|
-      #########################################
-      #   For render react static files
-      routing.public
 
-      #   GET /
+    route do |routing|
+      # GET /
       routing.root do
-        File.read('app/presentation/built/index.html')
-      end
+        message = "PaperDeep API v1 at /api/v1/ in #{App.environment} mode"
 
-      routing.on ['test2', 'citedResult'] do
-        File.read('app/presentation/built/index.html')
-      end
+        result_response = Representer::HttpResponse.new(
+          Response::ApiResult.new(status: :ok, message: message)
+        )
 
-      #########################################
-      #   For Apis
-      routing.on 'search' do
-        routing.is do
-          # POST /search/
-          routing.post do
-            params = JSON.parse(routing.body.read)
-            search_request = PaperDeep::Forms::NewSearch.new.call(params)
-            result = PaperDeep::Service::AddPaper.new.call(search_request)
-
-            if result.failure?
-              flash[:error] = result.failure
-              return { result: false, error: flash[:error] }.to_json
-            end
+        api_v1_link = [
+          search: 'api/v1/search',
+          citationtree: 'api/v1/citationtree',
+          publication: 'api/v1/publication',
+          db_eid: 'api/v1/db/eid'
+        ]
 
-            Views::Papers.new(result.value![:paper]).content.to_json
-          end
-        end
-        routing.on 'publication' do
+        full_response = JSON.parse(result_response.to_json)
+        full_response['link'] = api_v1_link
+
+        response.status = result_response.http_status_code
+        full_response.to_json
+      end
+      #   For Apis
+      routing.on 'api/v1' do
+        routing.on 'search' do
```

Comment:
In restful api suggestions, api route should not include verb. The action should be represented by the GET/POST. 
The route seems that it will get the related publication. You can name after what you get, e.g. papers.

Reply:
Or there is other way to do.
POST /papers/{study_field}

Reply:
Hi @chinyakao,

thanks for reminding us!
last week we just think search can also be a noun
```
POST /papers/{study_field}
```
seems nice too
we will fix the verb naming in our next Week13 pr

Reply:
Hi @chinyakao ,

#13 In this PR, we change the api name from 'search' to 'paper'.
Thanks for the suggestion

PS. Also include this week's work (setting github action)